### PR TITLE
Monorepo setup with shared types

### LINKED ISSUE #29: Zod input validation frontend and backend sync

### PR DESCRIPTION
**Pull request by donljh**
* Moving from `npm` and using `pnpm` as our package manager instead since they offering much better experience with workspaces. Some QOL changes, (1) `pnpm i` at the root to install all dependencies -- `pnpm` lifts up common dependencies to the root level, so no need to deal with `package-lock.json` at the individual subfolder level, only 1 `pnpm-lock.yaml` at the root level (2) `pnpm dev` at the root level now runs the `dev` script for frontend and question service

* Created a common types package under `common/shared-types`, we can put our Request / Response types here and share them between front and backend

* Standardise the location of the firebase json file and name under `.firebase`, which is `git`-ignored

### CODE REVIEW

File: services/question/.env.development.example
Comment (Cloud7050) [inline services/question/.env.development.example:1]: Appreciate the sample env file.

**Review by ys112 — APPROVED** [review]
Tested and working! LGTM

File: services/question/.env.development.example
Comment (donljh) [reply services/question/.env.development.example:1]: Possible to help approve the PR?

**Review by Cloud7050 — APPROVED** [review]

**Comment by Cloud7050** [comment]
Will take a closer look while merging into qcon branch, and update pending changes there accordingly.

**Pull request merged**